subsys/bluetooth/audio/pacs.c

```diff
@@ -246,13 +248,45 @@ static int set_available_contexts(uint16_t contexts, uint16_t *available,
 		return 0;
 	}
 
+	err = k_work_reschedule(&available_contexts_work, PAC_NOTIFY_TIMEOUT);
+	if (err < 0) {
+		return err;
+	}
+
 	*available = contexts;
 
-	err = k_work_reschedule(&available_contexts_work, PAC_NOTIFY_TIMEOUT);
+	return 0;
+}
+
+static int set_supported_contexts(uint16_t contexts, uint16_t *supported,
+				  uint16_t *available)
+{
+	int err;
+
+	/* Ensure unspecified is always supported */
+	contexts |= BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED;
+
+	if (*supported == contexts) {
+		return 0;
+	}
+
+	err = k_work_reschedule(&supported_contexts_work, PAC_NOTIFY_TIMEOUT);
 	if (err < 0) {
 		return err;
 	}
 
+	*supported = contexts;
+
+	/* Update available contexts if needed*/
+	if ((contexts & *available) != *available) {
+		*available = *available & contexts;
+		err = k_work_reschedule(&available_contexts_work,
+					PAC_NOTIFY_TIMEOUT);
```

Comment:
I honestly don't know if this is the right way, or if we should avoid setting `*supported` if this also fails. 

Maybe the assignment to `*supported` and `*available` shall only be done after both `k_work_reschedule` succeeds? And if we end up doing that, then if `k_work_reschedule(&available_contexts_work` fails, then we should also cancel the `k_work_reschedule(&supported_contexts_work`

On the other hand, since we are offloading this to a k_work, then `bt_gatt_notify_uuid` may still fail even if `k_work_reschedule` succeeds, and then we still can't be sure the notification is sent. In that regard, maybe we need to get rid of the `k_work` usage in this file too. @MariuszSkamra what do you think?

Reply:
@Thalley Indeed, probably we don't need to offload context notifications to system workqueue. I do not know why we do that actually, maybe for some historical reasons ;)

Reply:
> I do not know why we do that actually, maybe for some historical reasons ;)

Indeed for historical reasons :) 

Is it something we want to fix in this PR, or offload to a new PR?

Reply:
I'd prefer offloading this in new PR.

Reply:
Created https://github.com/zephyrproject-rtos/zephyr/issues/53775